content/docs/04-ai-sdk-ui/04-generative-ui.mdx

```diff
@@ -0,0 +1,347 @@
+---
+title: Generative UI
+description: Learn how to build Generative UI with AI SDK UI.
+---
+
+# Generative UI
+
+Generative UI is the process of allowing a large language model (LLM) to go beyond text and "generate UI". This creates a more engaging and AI-native experience for users.
+
+<WeatherSearch />
+
+At the core of generative UI are [ tools ](/docs/ai-sdk-core/tools-and-tool-calling), which are like programs you provide to the model to perform specialized tasks like getting the weather in a location. The model can decide when and how to use these tools based on the context of the conversation.
+
+Generative UI is simply the process of piping the results of a tool call to a React component. Here's how the process works:
+
+1. You provide the model with a prompt or conversation history, along with a set of tools.
+2. Based on the context, the model may decide to call a tool.
+3. If a tool is called, it will execute and return data.
+4. This data can then be piped to a React component for rendering.
+5. If no relevant tool is available or a tool isn't necessary (ie. summarising actions), the model will generate text.
```

Comment:
The model now always generates text at the end?

Reply:
maybe we just remove that last bullet then?